proto/p4/config/v1/p4info.proto

```diff
@@ -217,6 +217,9 @@ message Table {
   repeated ActionRef action_refs = 3;
   // 0 (default value) means that the table does not have a const default action
   uint32 const_default_action_id = 4;
+  // The initial default action of the table. This can be overridden at runtime.
+  // 0 if there is no default action.
```

Comment:
"0 if there is no default action".  I am confused by this.  Even for P4 tables with no explicit `default_action` property, the P4 language spec says that the default_action for such a table is `NoAction`, and in p4c today this is an explicit separate action in the P4Info file with a non-0 action id.

Reply:
It appears this sentence may have been copied from the comments describing const_default_action_id above, but note that it is definitely possible to define a table that does not include `const default_action = foo`, whereas I do not think it is possible to define a table that has no default_action at all.

Reply:
I have added the sentence because the current implementation allows for default actions to be missing from a table: https://github.com/p4lang/p4c/blob/main/control-plane/p4RuntimeSerializer.cpp#L258

Reply:
Is it actually possible to hit that case when executing that code, given the current front end compiler code?  I suspect the answer is "no".

Reply:
With the current front end I do not think so, but the front end is rather bloated and is likely to change: https://github.com/p4lang/p4c/issues/4634

I think it makes sense to throw an error here then.

Reply:
Even if the front end changes radically, this behavior of treating a table without a source-code-specified `default_action` as one with `default_action = NoAction` is codified in the language spec, so it is one aspect of the front end that I believe should _not_ change.

See Section 14.2: "The compiler must set the default_action to NoAction (and also insert it into the list of actions) for tables that do not define the default_action property.  Hence, all tables can be thought of
as having a default_action` property, either implicitly or explicitly."